code/modules/hydroponics/plant_genes.dm

```diff
@@ -465,116 +533,197 @@
-	if(istype(I, /obj/item/stack/cable_coil))
-		var/obj/item/stack/cable_coil/C = I
-		if(C.use(5))
-			to_chat(user, "<span class='notice'>You add some cable to [G] and slide it inside the battery encasing.</span>")
-			var/obj/item/stock_parts/cell/potato/pocell = new /obj/item/stock_parts/cell/potato(user.loc)
-			pocell.icon_state = G.icon_state
-			pocell.maxcharge = G.seed.potency * 20
-
-			// The secret of potato supercells!
-			var/datum/plant_gene/trait/cell_charge/CG = G.seed.get_gene(/datum/plant_gene/trait/cell_charge)
-			if(CG) // Cell charge max is now 40MJ or otherwise known as 400KJ (Same as bluespace power cells)
-				pocell.maxcharge *= CG.rate*100
-			pocell.charge = pocell.maxcharge
-			pocell.name = "[G.name] battery"
-			pocell.desc = "A rechargeable plant-based power cell. This one has a rating of [DisplayEnergy(pocell.maxcharge)], and you should not swallow it."
-
-			if(G.reagents.has_reagent(/datum/reagent/toxin/plasma, 2))
-				pocell.rigged = TRUE
-
-			qdel(G)
-		else
-			to_chat(user, "<span class='warning'>You need five lengths of cable to make a [G] battery!</span>")
+/datum/plant_gene/trait/battery/on_new(obj/item/our_plant, newloc)
+	if(!..())
+		return
+
+	RegisterSignal(our_plant, COMSIG_PARENT_ATTACKBY, .proc/make_battery)
 
+/*
+ * When a plant with this gene is hit (attackby) with cables, we turn it into a battery.
+ *
+ * our_plant - our plant being hit
+ * hit_item - the item we're hitting the plant with
+ * user - the person hitting the plant with an item
+ */
+/datum/plant_gene/trait/battery/proc/make_battery(obj/item/our_plant, obj/item/hit_item, mob/user)
+	SIGNAL_HANDLER
 
+	if(!istype(hit_item, /obj/item/stack/cable_coil))
+		return
+
+	var/obj/item/seeds/our_seed = our_plant.get_plant_seed()
+	var/obj/item/stack/cable_coil/cabling = hit_item
+	if(cabling.use(5))
+		to_chat(user, "<span class='notice'>You add some cable to [our_plant] and slide it inside the battery encasing.</span>")
+		var/obj/item/stock_parts/cell/potato/pocell = new /obj/item/stock_parts/cell/potato(user.loc)
+		pocell.icon_state = our_plant.icon_state
+		pocell.maxcharge = our_seed.potency * 20
+
+		// The secret of potato supercells!
+		var/datum/plant_gene/trait/cell_charge/electrical_gene = our_seed.get_gene(/datum/plant_gene/trait/cell_charge)
+		if(electrical_gene) // Cell charge max is now 40MJ or otherwise known as 400KJ (Same as bluespace power cells)
+			pocell.maxcharge *= electrical_gene.rate*100
+		pocell.charge = pocell.maxcharge
+		pocell.name = "[our_plant.name] battery"
+		pocell.desc = "A rechargeable plant-based power cell. This one has a rating of [DisplayEnergy(pocell.maxcharge)], and you should not swallow it."
+
+		if(our_plant.reagents.has_reagent(/datum/reagent/toxin/plasma, 2))
+			pocell.rigged = TRUE
+
+		qdel(our_plant)
+	else
+		to_chat(user, "<span class='warning'>You need five lengths of cable to make a [our_plant] battery!</span>")
```

Comment:
```suggestion
	if(!cabling.use(5))
		to_chat(user, "<span class='warning'>You need five lengths of cable to make a [our_plant] battery!</span>")
		return
	to_chat(user, "<span class='notice'>You add some cable to [our_plant] and slide it inside the battery encasing.</span>")
	var/obj/item/stock_parts/cell/potato/pocell = new /obj/item/stock_parts/cell/potato(user.loc)
	pocell.icon_state = our_plant.icon_state
	pocell.maxcharge = our_seed.potency * 20

	// The secret of potato supercells!
	var/datum/plant_gene/trait/cell_charge/electrical_gene = our_seed.get_gene(/datum/plant_gene/trait/cell_charge)
	if(electrical_gene) // Cell charge max is now 40MJ or otherwise known as 400KJ (Same as bluespace power cells)
		pocell.maxcharge *= electrical_gene.rate*100
	pocell.charge = pocell.maxcharge
	pocell.name = "[our_plant.name] battery"
	pocell.desc = "A rechargeable plant-based power cell. This one has a rating of [DisplayEnergy(pocell.maxcharge)], and you should not swallow it."

	if(our_plant.reagents.has_reagent(/datum/reagent/toxin/plasma, 2))
		pocell.rigged = TRUE

	qdel(our_plant)
```